File: java/Leccion7/JavaBeans/src/domain/Persona.java

package domain;

//Esta clase es de java beans
/*
Un JavaBean es una clase de Java que sigue ciertas convenciones de nomenclatura, 
estructura y comportamiento, lo que permite que sea utilizado de manera sencilla 
y consistente en entornos de desarrollo y herramientas.

Un JavaBean debe cumplir con las siguientes características:

-Debe tener una clase pública que siga las convenciones de nomenclatura de Java.
-Debe tener un constructor público sin argumentos.
-Los atributos deben ser privados y accedidos mediante métodos getter y setter.
-Los métodos getter y setter deben seguir las convenciones de nomenclatura de Java.
-Puede tener otros métodos para proporcionar funcionalidad adicional.
Los JavaBeans son utilizados principalmente en entornos de desarrollo visual 
y en tecnologías como JavaFX y Java Servlets. Proporcionan una forma estándar 
de encapsular datos y funcionalidad en un componente que puede ser utilizado 
y manipulado fácilmente por herramientas de desarrollo y bibliotecas de terceros.
*/

import java.io.Serializable;

public class Persona implements Serializable{
    private String nombre;
    private String apellido;
    
    //Es obligatorio que tenga un constructor vacio para que sea java beans

    public Persona() {
    }

    public String getNombre() {
        return this.nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getApellido() {
        return this.apellido;
    }

    public void setApellido(String apellido) {
        this.apellido = apellido;
    }

    @Override
    public String toString() {
        return "Persona{" + "nombre=" + nombre + ", apellido=" + apellido + '}';
    }
}

File: JavaScript/leccion7/persona.js
class Persona{
    static contadorPersona = 0;
    constructor(nombre, apellido, edad){
        this.id = ++Persona.contadorPersona;
        this._nombre = nombre;
        this._apellido = apellido;
        this._edad = edad;
    }
    get idPersona(){
        return this.id;
    }

    get nombre(){
        return this._nombre;
    }

    set nombre(nombre){
        this._nombre = nombre;
    }

    get apellido(){
        return this._apellido;
    }

    set apellido(apellido){
        this._apellido = apellido;
    }

    get edad(){
        return this._edad 
    }

    set edad(edad){
        this._edad = edad;
    }

    toString(){
        return this.id + ' ' + this._nombre + ' ' + this._apellido + ' ' + this._edad;
    }
}